Runtime/Scripts/RGBotLocalRuntime/RG.cs

```diff
@@ -0,0 +1,192 @@
+         */
+        public List<RGStateEntity> GetMyPlayers()
+        {
+            return this.FindPlayers(ClientId);
+        }
+
+        /**
+         * <summary>Used to find the closest Entity to the given position.</summary>
+         * <param name="objectType">{string | null} Search for entities of a specific type</param>
+         * <param name="position">{Vector3 | null} Position to search from.  If not passed, attempts to use the client's bot position in index 0.</param>
+         * <returns>{RGStateEntity} The closest Entity matching the search criteria, or null if none match.</returns>
+         */
+        [CanBeNull]
+        public RGStateEntity FindNearestEntity(string objectType = null, Vector3? position = null)
+        {
+            var result = this.FindEntities(objectType);
+
+            var pos = position ?? this.GetMyPlayers()[0].position ?? Vector3.zero;
+            
+            result.Sort((e1, e2) =>
+            {
+                var val = MathFunctions.DistanceSq(pos, e1.position ?? Vector3.zero) -
+                             MathFunctions.DistanceSq(pos, e1.position ?? Vector3.zero);
+                if (val < 0)
+                {
+                    return -1;
+                }
+                else if (val > 0)
+                {
+                    return 1;
+                }
+
+                return 0;
+            });
+
+            return result.Count>0 ? result[0] : null;
+        }
+
+        /**
+         * <summary>Used to find a list of entities from the game state.</summary>
+         * <param name="objectType">{string | null} Search for entities of a specific type</param>
+         * <returns>{List&lt;RGStateEntity&gt;} All entities with the given objectType, or all entities in the state if objectType is null.</returns>
+         */
+        public List<RGStateEntity> FindEntities(string objectType = null)
+        {
+            var gameState = _tickInfo.gameState;
+            if (gameState.Count == 0)
+            {
+                return new List<RGStateEntity>();
+            }
+            
+            // filter down to objectType Matches
+            var result = gameState.Values.Where(value =>
+            {
+                if (objectType != null && value.type != null)
+                {
+                    return objectType.Equals(value.type);
+                }
+                return true;
+            });
+
+            return result.ToList();
+        }
+
+        /**
+         * <summary>Used to find a list of players from the game state.</summary>
+         * <param name="clientId">{uint | null} Search for players owned by a specific clientId</param>
+         * <returns>{List&lt;RGStateEntity&gt;} All players owned by the given clientId, or all players in the state if client is null.</returns>
+         */
+        public List<RGStateEntity> FindPlayers(uint? clientId = null)
```

Comment:
For local bots, I have doubts if this will get used.  The one use case I considered was games where you control multiple characters as a single player (like you have a team of 5 characters you're controlling as 1 player).  In these situations I presumed a single bot may run them to have the same sort of cross character vision as a human player would.